ykrt/src/mt.rs

```diff
@@ -67,18 +68,19 @@ pub struct MT {
 impl MT {
     // Create a new meta-tracer instance. Arbitrarily many of these can be created, though there
     // are no guarantees as to whether they will share resources effectively or fairly.
-    pub fn new() -> Self {
-        Self {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
```

Comment:
You are OK with `dyn Error` here because this is initialisation code that should be called only once?

Reply:
It is only called once, but even if it wasn't, why would `dyn Error` be a problem?

Reply:
Its slower than a static error.

Reply:
Yes, I meant why don't you use a custom error type.

I thought that a trait object would be bigger than a word due to an additional vtable pointer?

Anyway, I don't feel strongly.

Reply:
> I thought that a trait object would be bigger than a word due to an additional vtable pointer?

`Box<dyn ...>` means the vtable pointer is held behind a `Box`, so the `Err` only holds a single machine word.